api/test/integration/test_auth.py

```diff
@@ -54,20 +53,19 @@ def test_auth_tokens_registration(client):
 
 @pytest.mark.django_db
 @pytest.fixture
-def test_auth_token_exchange(client, test_auth_tokens_registration):
-    client_id = test_auth_tokens_registration["client_id"]
-    client_secret = test_auth_tokens_registration["client_secret"]
-    data = urlencode(
-        {
-            "client_id": client_id,
-            "client_secret": client_secret,
-            "grant_type": "client_credentials",
-        }
-    )
-    res = client.post(
+def test_auth_token_exchange(api_client, test_auth_tokens_registration):
+    api_client_id = test_auth_tokens_registration["client_id"]
+    api_client_secret = test_auth_tokens_registration["client_secret"]
+    data = {
+        "client_id": api_client_id,
+        "client_secret": api_client_secret,
+        "grant_type": "client_credentials",
+    }
+
+    res = api_client.post(
         "/v1/auth_tokens/token/",
         data,
-        "application/x-www-form-urlencoded",
+        "multipart",
```

Comment:
What is the reason for this change?

Reply:
It's part of DRF's `api_client`'s testing request renderer formats. It takes a slug corresponding to the renderer rather than the content type string: https://www.django-rest-framework.org/api-guide/testing/#setting-the-available-formats

Here's an SO answer explaining it as well: https://stackoverflow.com/questions/68528408/difference-between-content-type-application-json-and-format-json-in-drf-apiclien

DRF's testing APIClient actually gave a pretty easy to understand error message about this, I was not aware of this until fixing that error when I changed this test to api_client.